src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilationBuilder.cs

```diff
@@ -61,9 +61,6 @@ public sealed class ReadyToRunCodegenCompilationBuilder : CompilationBuilder
             _ilProvider = new ReadyToRunILProvider(group);
             _inputFiles = inputFiles;
             _compositeRootPath = compositeRootPath;
-
-            // R2R field layout needs compilation group information
-            ((ReadyToRunCompilerContext)context).SetCompilationGroup(group);
```

Comment:
I've moved this so it happens right after we create the compilation group so it is early enough for the `ProfileDataManager` that also needs this now (for some `EnsureLoadable` calls).